imbi/endpoints/ssm/models.py

```diff
@@ -0,0 +1,111 @@
+import typing
+
+import pydantic
+
+
+class Param(pydantic.BaseModel):
+    type: typing.Literal['String', 'SecureString', 'StringList']
+    value: str
+
+
+class EnvironmentFieldMixin:
+    @pydantic.computed_field
+    @property
+    def environment(self) -> str:
+        return self.path.split('/')[1]
+
+
+class RemovePatch(pydantic.BaseModel, EnvironmentFieldMixin):
+    op: typing.Literal['remove']
+    path: str
+
+    @pydantic.field_validator('path', mode='after')
+    def validate_path(cls, value: str, info: pydantic.ValidationInfo):
+        if value.count('/') > 1:
+            raise ValueError(
+                f'Invalid path: {value}. Can only remove an environment')
+
+        environments = info.context.get('environments', [])
+        environment = value.split('/')[1]
+        if environment not in environments:
+            raise ValueError(
+                f'Invalid environment "{environment}", must be one of: '
+                f'{environments}')
+
+        return value
+
+
+class ReplacePatch(pydantic.BaseModel, EnvironmentFieldMixin):
+    op: typing.Literal['replace']
+    path: str
+    value: str
+
+    @pydantic.computed_field
+    @property
+    def type(self) -> str:
+        return self.path.split('/')[2]
+
+    @pydantic.model_validator(mode='after')
+    def validate_replace(self, info: pydantic.ValidationInfo):
+        environments = info.context.get('environments', [])
+
+        if self.path.count('/') != 2:
+            raise ValueError(
+                f'Invalid path: {self.value}. Must replace type or value at '
+                'an environment')
+
+        _, environment, update_type = self.path.split('/')
```

Comment:
The following is equivalent and _slightly_ more intentional
```suggestion
try:
  _, environment, update_type = self.path.split('/')
except ValueError:
  raise ValueError(
    f'Invalid path: {self.value}. Must replace type or value at '
    'an environment')
```